File: frontend/src/types/classTrends.ts
// Mirrors backend/app/schemas/class_trends.py

export interface RosterAccuracyTrendPoint {
  date: string;
  attempts: number;
  scored_attempts: number;
  accuracy_percent: number | null;
}

export interface RosterCourseCompletion {
  course_id: string;
  course_title: string;
  learner_count: number;
  not_started_count: number;
  in_progress_count: number;
  completed_count: number;
  certified_count: number | null;
}

export interface ClassTrendsResponse {
  accuracy_trend: RosterAccuracyTrendPoint[];
  course_completion: RosterCourseCompletion[];
}

File: backend/app/schemas/class_trends.py
from typing import Optional

from pydantic import BaseModel


class RosterAccuracyTrendPoint(BaseModel):
    date: str
    attempts: int
    scored_attempts: int
    accuracy_percent: Optional[float] = None


class RosterCourseCompletion(BaseModel):
    course_id: str
    course_title: str
    learner_count: int
    not_started_count: int
    in_progress_count: int
    completed_count: int
    certified_count: Optional[int] = None


class ClassTrendsResponse(BaseModel):
    accuracy_trend: list[RosterAccuracyTrendPoint]
    course_completion: list[RosterCourseCompletion]
